cernopendata/cold_storage/api.py

```diff
@@ -0,0 +1,396 @@
+            if limit > 0:
+                transfers = Request.query.filter_by(
+                    status="submitted", action=action
+                ).all()
+
+                for transfer in transfers:
+                    info = manager.doOperation(
+                        action,
+                        transfer.record_id,
+                        limit=limit,
+                        register=True,
+                        force=False,
+                        dry=False,
+                    )
+                    logger.debug(f"Got {info}")
+                    if info:
+                        submitted += len(info)
+                        transfer.num_files += len(info)
+                        transfer.size += sum(item.size for item in info)
+                    transfer.started_at = datetime.utcnow()
+                    if limit == submitted:
+                        logger.info(
+                            f"Reached the threshold of {threshold} transfers. There might be more in this record"
+                            f"({submitted + active_transfers_count}). Let's wait before continuing"
+                        )
+                    else:
+                        transfer.status = "started"
+                    db.session.add(transfer)
+                    limit -= submitted
+                    if limit <= 0:
+                        logger.info("We have submitted enough. Stopping")
+                        break
+            if submitted:
+                logger.info(f"{submitted} transfers have been submitted!")
+            db.session.commit()
+
+    @staticmethod
+    def _check_running():
+        """Check the records that are being archived."""
+        for action in ["stage", "archive"]:
+
+            requests = Request.query.filter_by(status="started", action=action).all()
+            logger.debug(f"Checking the {len(requests)} {action} requests")
+            completed = 0
+            for request in requests:
+                record = RecordFilesWithIndex.get_record(request.record_id)
+
+                if action == "stage":
+                    if record["availability"] != "online":
+                        continue
+                elif action == "archive":
+                    files = (f for index in record.file_indices for f in index["files"])
+                    missing = next(
+                        (
+                            f
+                            for f in files
+                            if "tags" not in f or "uri_cold" not in f["tags"]
+                        ),
+                        None,
+                    )
+                    if missing:
+                        logger.debug(f"The file {missing['key']} is not in tape yet...")
+                        continue
+                completed += 1
+                RequestAPI.complete(request)
+            logger.info(f"{completed}/{len(requests)} requests have finished")
+
+    @staticmethod
+    def get_requests(
```

Comment:
```suggestion
    def search_request(
```
(and move it to the manager/service)

Reply:
done